lib/rpmi_service_group_syssusp.c

```diff
@@ -172,19 +172,19 @@ static enum rpmi_error rpmi_syssusp_process_events(struct rpmi_service_group *gr
 
 	switch (sgsusp->current_state) {
 	case RPMI_SYSSUSP_STATE_SUSPEND_PENDING:
-		if (!sgsusp->ops->system_suspend_ready(sgsusp->ops_priv,
-						       sgsusp->current_hart_index))
-			return RPMI_SUCCESS;
+		if (sgsusp->ops->system_suspend_ready(sgsusp->ops_priv,
+						      sgsusp->current_hart_index))
+			return RPMI_ERR_BUSY;
```

Comment:
This looks incorrect because we should be doing nothing when system is not ready to suspend.